src/main/java/io/ebean/dbmigration/ddlgeneration/platform/PlatformDdl.java

```diff
@@ -219,7 +222,7 @@ public String convertDefaultValue(String dbDefault) {
    * Return the drop foreign key clause.
    */
   public String alterTableDropForeignKey(String tableName, String fkName) {
-    return "alter table " + alterTableIfExists + tableName + " " + dropConstraintIfExists + " " + fkName;
+    return "alter table " + alterTableIfExists + tableName + " " + dropConstraintIfExists + " " + maxConstraintName(fkName);
```

Comment:
I had problems with the constraint name limit on oracle. First I thought, that ebean already shortens the names, but as far as I understand the dbmigration process, table names are shortened during the deploy phase and the BeanDescriptor has already the shortened names. But this happens with the current platfrom, which is H2(offline)
So the names in the model.xml's are too long for oracle & db2. This is easy to fix by specifying a shorter name with a @Table() annotation.

Unfortunately foreignkeys of m2m tables are rather long, as the m2m table name = table1+table2 and the fk is m2mTableName+table1+propName which gets quite huge: 'fk_migtest_mtm_c_migtest_mtm_m_migtest_mtm_m'

So I shorten the names here.